### PR TITLE
Small cleanup of client cache related classes

### PR DESCRIPTION
**Pull request by Donnerbart**
No logic has been changed, just warnings removed, JavaDoc improvements, style alignments.

### CODE REVIEW

File: hazelcast-client/src/main/java/com/hazelcast/client/cache/impl/AbstractClientCacheProxy.java [hunk 1]
```diff
@@ -429,11 +430,11 @@ private void waitResponseFromAllPartitionsForPutAll(List<FutureEntriesTuple> fut
              *        In this test exception is thrown at `CacheWriter` and caller side expects this exception.
              * So as a result, we only throw the first exception and others are suppressed by only logging.
              */
-            ExceptionUtil.rethrow(error);
+            throw rethrow(error);
         }
     }
 
-    private void handleNearCacheOnPutAll(List<Map.Entry<Data, Data>> entries, boolean invalidate) {
```
Comment (pveentjer) [inline hazelcast-client/src/main/java/com/hazelcast/client/cache/impl/AbstractClientCacheProxy.java]: What happened tot the invalidate flag? Not needed anymore?

Comment (Donnerbart) [inline hazelcast-client/src/main/java/com/hazelcast/client/cache/impl/AbstractClientCacheProxy.java]: It was an unused parameter on a private method. The whole method was empty until November 2015, but it looks like the parameter was never used.

**Comment by Donnerbart** [comment]
run-lab-run

File: hazelcast-client/src/main/java/com/hazelcast/client/cache/impl/AbstractClientCacheProxyBase.java [hunk 2]
```diff
@@ -174,9 +177,7 @@ public void open() {
         if (isDestroyed.get()) {
             throw new IllegalStateException("Cache is already destroyed! Cannot be reopened");
         }
-        if (!isClosed.compareAndSet(true, false)) {
-            return;
-        }
+        isClosed.compareAndSet(true, false);
```
Comment (pveentjer) [inline hazelcast-client/src/main/java/com/hazelcast/client/cache/impl/AbstractClientCacheProxyBase.java]: Why not call set(false)?

Comment (Donnerbart) [inline hazelcast-client/src/main/java/com/hazelcast/client/cache/impl/AbstractClientCacheProxyBase.java]: Done

**Pull request merged**